internal/agent/config.go

```diff
@@ -117,7 +117,7 @@ func (cfg *Config) Validate() error {
 		{cfg.ConfigDir, "config-dir", true},
 		{cfg.DataDir, "data-dir", true},
 		{cfg.Cacert, "ca-cert", true},
-		{cfg.Key, "key", true},
+		{cfg.Key, "key", false},
```

Comment:
This still has me slightly confused. In the production ENV we are getting the key from what again? Today the agent will create a new keypair if it doesn't exist but that should be removed IMO. @fzdarsky has explained it but we should document it.

Reply:
The agent creates the agent.key at boot, eventually requests the TPM to create a key instead..

Reply:
So it should not exist... it's itself who will create it. I wonder how are the tests passing ':D may be because we don't call validate on the config, and just start the Agent controller from tests?

Reply:
> The agent creates the agent.key at boot, eventually requests the TPM to create a key instead..
OK maybe we should move this logic into the mock TMP.

Reply:
> it's itself who will create it.

It makes sense that the TPM will generate it. I wasn't certain on the management of the keys.